lib/dbservice_web/controllers/student_controller.ex

```diff
@@ -390,4 +393,171 @@ defmodule DbserviceWeb.StudentController do
     )
     |> Repo.exists?()
   end
+
+  swagger_path :create_student_id do
+    post("/api/student/generate-id")
+
+    parameters do
+      body(:body, Schema.ref(:StudentIdGeneration), "Details for generating student ID",
+        required: true
+      )
+    end
+
+    response(201, "Created", Schema.ref(:StudentIdGenerationResponse))
+  end
+
+  def create_student_id(conn, params) do
+    try do
+      case generate_student_id(params) do
+        "" ->
+          conn
+          |> put_status(:bad_request)
+          |> json(%{error: "Could not create student id"})
+
+        student_id ->
+          conn
+          |> put_status(:created)
+          |> json(%{student_id: student_id})
+      end
+    rescue
+      e in RuntimeError ->
+        conn
+        |> put_status(:internal_server_error)
+        |> json(%{error: e.message})
+
+      e in _ ->
+        conn
+        |> put_status(:internal_server_error)
+        |> json(%{error: "An unexpected error occurred: #{inspect(e)}"})
+    end
+  end
+
+  defp generate_student_id(params) do
+    grade = Grades.get_grade_by_params(%{number: params["grade"]})
+
+    existing_students = get_existing_students(grade.id, params["category"])
+
+    case find_existing_student_id(existing_students, params) do
+      "" ->
+        generate_new_student_id(params)
+
+      student_id ->
+        student_id
+    end
+  end
+
+  defp get_existing_students(grade_id, category) do
+    Users.get_students_by_params(%{grade_id: grade_id, category: category})
+  end
+
+  defp find_existing_student_id(existing_students, params) do
+    Enum.find_value(existing_students, "", fn existing_student ->
+      existing_user = get_existing_user(existing_student.user_id, params)
+
+      check_enrollment_and_get_id(existing_user, existing_student.student_id, params)
```

Comment:
Why are we checking enrollment and all, and not just the existence of student id/

Reply:
Based on the JNV ID logic in Portal Backend, we are checking if the user is enrolled in the given school.